slack-moderate-words/events.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	"sigs.k8s.io/slack-infra/slack"
+	"sigs.k8s.io/slack-infra/slack-moderate-words/model"
+)
+
+type handler struct {
+	client  *slack.Client
+	filters model.FilterConfig
+}
+
+// ServeHTTP handles Slack webhook requests.
+func (h *handler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		logError(rw, "Failed to read incoming request body: %v", err)
+		return
+	}
+	defer r.Body.Close()
+
+	event := &model.SlackEvent{}
+	err = json.NewDecoder(bytes.NewReader(body)).Decode(event)
+	if err != nil {
+		logError(rw, "Failed to unmarshal payload: %v", err)
+		panic(err)
+	}
+
+	if event.Type == "url_verification" {
```

Comment:
Should we `VerifySignature` before this?

https://github.com/kubernetes-sigs/slack-infra/blob/1eb97f5f2eb25163f26aba3e9ac6894b6a7c0133/slack-moderator/handler.go#L41

Reply:
done, thanks for the review